pkg/userdata/generate/generate.go

```diff
@@ -64,28 +65,53 @@ type TrustdInfo struct {
 	Token string
 }
 
-// RandomString returns a string of length n.
-func RandomString(n int) string {
-	var letter = []rune("abcdefghijklmnopqrstuvwxy0123456789")
+// randBytes returns a random string consisting of the characters in
+// validBootstrapTokenChars, with the length customized by the parameter
+func randBytes(length int) string {
 
-	b := make([]rune, n)
-	for i := range b {
-		b[i] = letter[rand.Intn(len(letter))]
+	// validBootstrapTokenChars defines the characters a bootstrap token can consist of
+	const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+	// len("0123456789abcdefghijklmnopqrstuvwxyz") = 36 which doesn't evenly divide
+	// the possible values of a byte: 256 mod 36 = 4. Discard any random bytes we
+	// read that are >= 252 so the bytes we evenly divide the character set.
+	const maxByteValue = 252
+
+	var (
+		b     byte
+		err   error
+		token = make([]byte, length)
+	)
+
+	reader := bufio.NewReaderSize(rand.Reader, length*2)
+	for i := range token {
+		for {
+			if b, err = reader.ReadByte(); err != nil {
+				return ""
+			}
+			if b < maxByteValue {
+				break
```

Comment:
looks like this could be a bit simplified if we drop inner `for` and use `continue` here instead of `break`?

Reply:
yep, good call. this was some shamelessly borrowed code from upstream k8s. but I def like that better.